Source/WebKit/UIProcess/Automation/BidiStorageAgent.cpp

```diff
@@ -0,0 +1,302 @@
+    auto partitionKey = parsedPartitionKey.value();
+
+    auto parsedCookieStore = cookieStoreForPartition(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedCookieStore, InternalError);
+    Ref<API::HTTPCookieStore> resolvedCookieStore = parsedCookieStore.value();
+
+    if (!optionalFilter)
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(InvalidParameter);
+
+
+    resolvedCookieStore->cookies([callback = WTFMove(callback), filter = WTFMove(optionalFilter), partitionKey = WTFMove(partitionKey)](Vector<WebCore::Cookie>&& cookiesList) mutable {
+        Vector<WebCore::Cookie> matchingCookies;
+        matchingCookies.reserveInitialCapacity(cookiesList.size());
+
+        for (const auto& cookie : cookiesList) {
+            if (cookieMatchesFilter(cookie, filter))
+                matchingCookies.append(cookie);
+        }
+
+        callback({ { buildArrayForCookies(matchingCookies), partitionKey } });
+    });
+
+}
+
+void BidiStorageAgent::setCookie(Ref<JSON::Object>&& cookie, RefPtr<JSON::Object>&& optionalPartition, Inspector::CommandCallback<Ref<PartitionKey>>&& callback)
+{
+    RefPtr session = m_session.get();
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!session, InternalError);
+
+    auto parsedPartitionKey = makePartitionKey(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedPartitionKey, InternalError);
+    auto partitionKey = parsedPartitionKey.value();
+
+    auto parsedCookieStore = cookieStoreForPartition(optionalPartition);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!parsedCookieStore, InternalError);
+    auto cookieStore = parsedCookieStore.value();
+
+    String name = cookie->getString("name"_s);
+    String value = cookie->getString("value"_s);
+    String domain = cookie->getString("domain"_s);
+
+    if (name.isEmpty() || value.isEmpty() || domain.isEmpty()) {
+        LOG(Automation, "setCookie: Missing required fields (name, value, domain)");
+        return;
+    }
+
+    WebCore::Cookie webCoreCookie;
+    webCoreCookie.name = name;
+    webCoreCookie.value = value;
+    webCoreCookie.domain = domain;
+    webCoreCookie.path = cookie->getString("path"_s);
+
+    auto secureOptional = cookie->getBoolean("secure"_s);
+    webCoreCookie.secure = secureOptional.value_or(false);
+
+    auto httpOnlyOptional = cookie->getBoolean("httpOnly"_s);
+    webCoreCookie.httpOnly = httpOnlyOptional.value_or(false);
+
+    Vector<WebCore::Cookie> cookiesToSet;
+    cookiesToSet.append(WTFMove(webCoreCookie));
+
+    cookieStore->setCookies(WTFMove(cookiesToSet), [callback = WTFMove(callback), partitionKey = WTFMove(partitionKey)]() mutable {
+        callback({ WTFMove(partitionKey) });
+    });
+};
+
+static void deleteCookiesSequentially(API::HTTPCookieStore* store, Vector<WebCore::Cookie> cookies, size_t index, Ref<PartitionKey> partitionKey, Inspector::CommandCallback<Ref<PartitionKey>> callback)
+{
+    if (index >= cookies.size()) {
+        LOG(Automation, "deleteCookies all deletions complete.");
```

Comment:
Since there's only one way to get to this code, I don't think this logging is adding anything. We can infer that the deletions are done if the `deleteCookies` command had returned.